server/controllers/CommentController.js

```diff
@@ -0,0 +1,190 @@
+        res.status(201).json({ comment });
+      })
+      .catch(next);
+  }
+
+  /**
+   * Create a reply
+   * @param {Object} req - request object
+   * @param {Object} res - response object
+   * @param {Object} next - a call to the next function
+   * @returns {Object} created reply on success and error message on error
+   */
+  static createReply(req, res, next) {
+    const { body: commentBody } = req.body.comment;
+    const { slug, id: parentId } = req.params;
+    const { id, username, image } = req.user;
+    Article.findOne({ where: { slug } })
+      .then((article) => {
+        errorHelper.throwError(article, 'Article not found', 404);
+        return article.createComment({
+          body: commentBody.trim(),
+          ownerId: id,
+          type: 'reply',
+          parentId,
+          deleted: false,
+        });
+      })
+      .then(({ dataValues }) => {
+        const comment = Object
+          .assign(dataValues, { owner: { username, image } });
+        res.status(201).json({ comment });
+      })
+      .catch(next);
+  }
+
+  /**
+   * update a comment
+   * @param {Object} req - request object
+   * @param {Object} res - response object
+   * @param {Object} next - a call to the next function
+   * @returns {Object} updated on success and error message on error
+   */
+  static updateComment(req, res, next) {
+    const { body: commentBody } = req.body.comment;
+    const { id } = req.params;
+    const { id: ownerId } = req.user;
+    const updateFields = [];
+    if (commentBody && commentBody.trim()) updateFields.push('body');
+    Comment.update({ body: commentBody }, {
+      where: { id, ownerId },
+      attributes: {
+        exclude: ['authorId'],
+      },
+      fields: updateFields,
+      returning: true,
+    })
+      .then(([updated, comment]) => {
+        if (!updated) {
+          return res.status(404)
+            .json({
+              errors: {
+                message: 'Comment not found, or you do '
+                  + 'not have permission to update this comment',
+              },
+            });
+        }
+        return res.status(200).json({ comment: comment[0] });
+      })
+      .catch(next);
+  }
+
+  /**
+   * Get all the comments of an article
+   * @param {Object} req - request object
+   * @param {Object} res - response object
+   * @param {Object} next - a call to the next function
+   * @returns {Object} an array of comments
+   * on success and error message on error
+   */
+  static getComments(req, res, next) {
+    const { slug } = req.params;
+    Article.findOne({ where: { slug } })
+      .then((article) => {
+        errorHelper.throwError(article, 'Article not found', 404);
+        const articleId = article.id;
+        // query the database to retrieve comments alongside their reply count
+        return sequelize
+          .query(queryHelper.getComments(articleId), {
+            type: sequelize.QueryTypes.SELECT,
```

Comment:
Dont you think it would be nice to include `pagination` to this query in order to avoid returning 1,000 comments at once